src/Control/Carrier/Class.hs

```diff
@@ -12,13 +12,18 @@ import Control.Effect.Reader (Reader(..))
 import Control.Effect.Sum ((:+:)(..))
 import Control.Effect.Writer (Writer(..))
 import Control.Monad ((<=<))
+import Data.List.NonEmpty (NonEmpty)
+import qualified Data.Semigroup as S
 
 -- | The class of carriers (results) for algebras (effect handlers) over signatures (effects), whose actions are given by the 'eff' method.
 class (HFunctor sig, Monad m) => Carrier sig m | m -> sig where
   -- | Construct a value in the carrier for an effect signature (typically a sum of a handled effect and any remaining effects).
   eff :: sig m a -> m a
 
 
+instance Carrier Choose NonEmpty where
+  eff (Choose m) = m True S.<> m False
```

Comment:
The qualified import is the nicest way to handle the fact that GHC 8.2 doesn’t have `<>` in scope, without mucking up an import with `#ifdef` or causing warnings on other compiler versions.

Reply:
Could also say `mappend` for that purpose, but I’m not particularly fussed.

Reply:
NonEmpty isn’t a `Monoid`!

Reply:
Oh wait is `mappend` a synonym for `<>` now?

Reply:
Oh of course it isn’t! My bad.